tuf/signed/sign_test.go

```diff
@@ -270,3 +248,59 @@ func TestSignWithX509(t *testing.T) {
 	assert.Len(t, testData.Signatures, 1)
 	assert.Equal(t, tufRSAx509Key.ID(), testData.Signatures[0].KeyID)
 }
+
+func TestSignWithOldKey(t *testing.T) {
+	cs := NewEd25519()
+	testData := data.Signed{}
+
+	key1, err := cs.Create(data.CanonicalRootRole, data.ED25519Key)
+	assert.NoError(t, err)
+
+	Sign(cs, &testData, key1)
+	assert.Len(t, testData.Signatures, 1)
+	assert.Equal(t, key1.ID(), testData.Signatures[0].KeyID)
+
+	key2, err := cs.Create(data.CanonicalRootRole, data.ED25519Key)
+	assert.NoError(t, err)
+
+	// should remove key1 sig even though it's valid. It no longer appears
+	// in the list of signing keys for the role
+	Sign(
+		cs,
+		&testData,
+		key2,
+	)
+
+	assert.Len(t, testData.Signatures, 1)
+	assert.Equal(t, key2.ID(), testData.Signatures[0].KeyID)
+}
+
+func TestSignWithOldSig(t *testing.T) {
```

Comment:
Nitpick:  could we rename this to something like `TestSignRemovesInvalidSignatures`?